make/ide/eclipse/CreateWorkspace.gmk

```diff
@@ -0,0 +1,392 @@
+              <entry flags="LOCAL" kind="includePath" name="$$(call FixPath, $$(call FindJavaHeaderDir, $$(module)))"/> \
+          </resource>$$(NEWLINE))))
+    endif
+
+    $$(eval $$(call SetupTextFileProcessing, $1_CREATE_SETTINGS_FILE, \
+      SOURCE_FILES := $(TOPDIR)/make/ide/eclipse/settings.template, \
+      OUTPUT_FILE := $$($1_SETTINGS_FILE), \
+      REPLACEMENTS := \
+          @@WORKSPACE@@ => $$($1_WORKSPACE_MAJOR) ; \
+          @@CSETTINGS@@ => $$($1_CSETTINGS) ; \
+          @@CXXSETTINGS@@ => $$($1_CXXSETTINGS), \
+    ))
+
+    TARGETS += $$($1_CREATE_SETTINGS_FILE)
+
+  endif
+
+  ifeq ($$($1_SHARED), false)
+    ifneq ($$(findstring $$($1_NATURE), JAVA MIXED), )
+      $1_LINKED_RESOURCES += $$(foreach src,$(JAVA_DIRS), \
+        <link> \
+            <name>$$(strip $$(src))</name> \
+            <type>2</type> \
+            <location>$$(subst \\,/,$$(call FixPath, $(TOPDIR)/$$(src)))</location> \
+        </link>$$(NEWLINE))
+    endif
+
+    ifneq ($$(findstring $$($1_NATURE), HOTSPOT NATIVE MIXED), )
+      # CDT includes a Makefile editor, so if CDT support was indicated there's no harm in
+      # including the make sources as well, since the cost of indexing them is miniscule
+      $1_LINKED_RESOURCES += \
+        <link> \
+            <name>make</name> \
+            <type>2</type> \
+            <location>$$(subst \\,/,$$(call FixPath, $(TOPDIR)/make))</location> \
+        </link> \
+        <link> \
+            <name>src/hotspot</name> \
+            <type>2</type> \
+            <location>$$(subst \\,/,$$(call FixPath, $(TOPDIR)/src/hotspot))</location> \
+        </link>$$(NEWLINE)
+    endif
+
+    ifneq ($$(findstring $$($1_NATURE), NATIVE MIXED), )
+      $1_LINKED_RESOURCES += $$(foreach src,$(NATIVE_DIRS), \
+        <link> \
+            <name>$$(strip $$(src))</name> \
+            <type>2</type> \
+            <location>$$(subst \\,/,$$(call FixPath, $(TOPDIR)/$$(src)))</location> \
+        </link>$$(NEWLINE))
+    endif
+  endif
+
+  $$(eval $$(call SetupTextFileProcessing, $1_CREATE_WORKSPACE_FILE, \
+    SOURCE_FILES := $(TOPDIR)/make/ide/eclipse/workspace.template, \
+    OUTPUT_FILE := $$($1_WORKSPACE_FILE), \
+    REPLACEMENTS := \
+        @@BUILD_MANAGERS@@ => $$($1_BUILD_MANAGERS) ; \
+        @@NATURES@@ => $$($1_NATURES) ; \
+        @@LINKED_RESOURCES@@ => $$($1_LINKED_RESOURCES), \
+  ))
+
+  TARGETS += $$($1_CREATE_WORKSPACE_FILE)
+
+endef
+
+ifeq ($(WORKSPACE), java)
+$(eval $(call SetupEclipseWorkspace, SETUP_WORKSPACE, \
+  NATURE := JAVA, \
+  SHARED := $(SHARED), \
+))
+else ifeq ($(WORKSPACE), hotspot)
+$(eval $(call SetupEclipseWorkspace, SETUP_WORKSPACE, \
+  NATURE := HOTSPOT, \
+  SHARED := $(SHARED), \
+))
+else ifeq ($(WORKSPACE), native)
+$(eval $(call SetupEclipseWorkspace, SETUP_WORKSPACE, \
+  NATURE := NATIVE, \
+  SHARED := $(SHARED), \
+))
+else
+$(eval $(call SetupEclipseWorkspace, SETUP_WORKSPACE, \
+  NATURE := MIXED, \
+  SHARED := $(SHARED), \
+))
+endif
```

Comment:
The lack of indentation makes this hard to read. This is how I would like the first block to look:
```
ifeq ($(WORKSPACE), java)
  $(eval $(call SetupEclipseWorkspace, SETUP_WORKSPACE, \
      NATURE := JAVA, \
      SHARED := $(SHARED), \
  ))
else ...
```